### PR TITLE
AC-103 VPC Private Gateways Bypass VLAN Overlap Check

### PR DESCRIPTION
**Pull request by nlgordon**
Adding the ability for VPC private gateways to overlap the normal VLAN ranges. By default VPC private gateway networks are of type Isolated. So I had to pull out most of the checks around the offering guest type being Isolated for overlaps.

### CODE REVIEW

File: engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java [hunk 1]
```diff
@@ -2194,13 +2194,13 @@ public Network createGuestNetwork(final long networkOfferingId, final String nam
         if (vlanSpecified) {
             URI uri = BroadcastDomainType.fromString(vlanId);
             //don't allow to specify vlan tag used by physical network for dynamic vlan allocation
```
Comment (Slair1) [inline engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java]: This comment is no longer valid, we should change to:

// Allow bypass of VLAN ID check for dynamic VLAN allocation ranges

File: engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java [hunk 2]
```diff
@@ -2194,13 +2194,13 @@ public Network createGuestNetwork(final long networkOfferingId, final String nam
...
-            if (!hasGuestBypassVlanOverlapCheck(bypassVlanOverlapCheck, ntwkOff) && _dcDao.findVnet(zoneId, pNtwk.getId(), BroadcastDomainType.getValue(uri)).size() > 0) {
+            if (!bypassVlanOverlapCheck && _dcDao.findVnet(zoneId, pNtwk.getId(), BroadcastDomainType.getValue(uri)).size() > 0) {
                 throw new InvalidParameterValueException("The VLAN tag " + vlanId + " is already being used for dynamic vlan allocation for the guest network in zone "
                         + zone.getName());
             }
             if (! UuidUtils.validateUUID(vlanId)){
                 // For Isolated and L2 networks, don't allow to create network with vlan that already exists in the zone
```
Comment (Slair1) [inline engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java]: This comment is no longer valid either.  Could change to:

// Allow bypass of VLAN ID check for Isolated and L2 networks

**Review by Slair1 — CHANGES_REQUESTED** [review]
I think we just need to update a couple out-of-date comments

File: engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java (hunk 1 above)
Comment (nlgordon) [reply engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java]: Fixed

File: engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java (hunk 2 above)
Comment (nlgordon) [reply engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java]: Fixed

**Review by Slair1 — APPROVED** [review]

**Pull request merged**